lib/peep/storage/striped.ex

```diff
@@ -0,0 +1,222 @@
+    tid = get_tid(tids)
+    key = {metric, tags}
+    :ets.update_counter(tid, key, {2, value}, {key, 0})
+  end
+
+  def insert_metric(tids, %Metrics.LastValue{} = metric, value, %{} = tags) do
+    tid = get_tid(tids)
+    now = System.monotonic_time()
+    key = {metric, tags}
+    :ets.insert(tid, {key, {now, value}})
+  end
+
+  def insert_metric(tids, %Metrics.Distribution{} = metric, value, %{} = tags) do
+    tid = get_tid(tids)
+    key = {metric, tags}
+
+    atomics =
+      case :ets.lookup(tid, key) do
+        [{_key, ref}] ->
+          ref
+
+        [] ->
+          # Race condition: Multiple processes could be attempting
+          # to write to this key. Thankfully, :ets.insert_new/2 will break ties,
+          # and concurrent writers should agree on which :atomics object to
+          # increment.
+          new_atomics = Storage.Atomics.new(metric)
+
+          case :ets.insert_new(tid, {key, new_atomics}) do
+            true ->
+              new_atomics
+
+            false ->
+              [{_key, atomics}] = :ets.lookup(tid, key)
+              atomics
+          end
+      end
+
+    Storage.Atomics.insert(atomics, value)
+  end
+
+  defp get_tid(tids) do
+    scheduler_id = :erlang.system_info(:scheduler_id)
+    %{^scheduler_id => tid} = tids
+    tid
+  end
+
+  def get_metric(tids, metrics, tags) when is_list(tags),
+    do: get_metric(tids, metrics, Map.new(tags))
+
+  def get_metric(tids, %Metrics.Counter{} = metric, tags) do
+    key = {metric, tags}
+
+    for tid <- Map.values(tids), reduce: 0 do
+      acc ->
+        case :ets.lookup(tid, key) do
+          [] -> acc
+          [{_, value}] -> acc + value
+        end
+    end
+  end
+
+  def get_metric(tids, %Metrics.Sum{} = metric, tags) do
+    key = {metric, tags}
+
+    for tid <- Map.values(tids), reduce: 0 do
+      acc ->
+        case :ets.lookup(tid, key) do
+          [] -> acc
+          [{_, value}] -> acc + value
+        end
+    end
+  end
+
+  def get_metric(tids, %Metrics.LastValue{} = metric, tags) do
+    key = {metric, tags}
+
+    {_ts, value} =
+      for tid <- Map.values(tids), reduce: nil do
+        acc ->
+          case :ets.lookup(tid, key) do
+            [] ->
+              acc
+
+            [{_, {_, _} = b}] ->
+              if acc do
+                max(acc, b)
+              else
+                b
+              end
+          end
+      end
+
+    value
+  end
+
+  def get_metric(tids, %Metrics.Distribution{} = metric, tags) do
+    key = {metric, tags}
+
+    merge_fun = fn _k, v1, v2 -> v1 + v2 end
+
+    for tid <- Map.values(tids), reduce: nil do
+      acc ->
+        case :ets.lookup(tid, key) do
+          [] ->
+            acc
+
+          [{_key, atomics}] ->
+            values = Storage.Atomics.values(atomics)
+
+            if acc do
+              Map.merge(acc, values, merge_fun)
+            else
+              values
+            end
```

Comment:
Might be a matter of taste but I'm not a fan of using `if` to check if something is nil or not.